### ISSUE TITLE
clap_plugin.reset() support

### ISSUE THREAD

**Issue by colugomusic** [iPlug2/iPlug2#1187]
Hello,

CLAP hosts rely on plugin developers to implement this function in order to revert plugins back to a "clean" DSP state. An example of where this is needed would be a DAW "panic" button which kills lingering reverb tails, delay feedback, etc.
https://github.com/free-audio/clap/blob/main/include/clap/plugin.h
```c
typedef struct clap_plugin {
   ...
   // - Clears all buffers, performs a full reset of the processing state (filters, oscillators,
   //   envelopes, lfo, ...) and kills all voices.
   // - The parameter's value remain unchanged.
   // - clap_process.steady_time may jump backward.
   //
   // [audio-thread & active]
   void(CLAP_ABI *reset)(const struct clap_plugin *plugin);
```
The vast majority of CLAP plugins don't bother to implement this. Before I begin my bug reporting crusade I first want to make sure all the most common CLAP plugin frameworks actually support it, and make sure that I know exactly what code changes to suggest for each plugin framework.

I can see that `IPlugCLAP` extends `clap::helpers::Plugin` so it is a case of overriding the virtual `reset()` function and calling into the plugin's reset code, wherever it is.

If I understand right then `IPlugCLAP` is directly extended by the plugin developer so they can simply override `clap::helpers::Plugin::reset` to fix their plugin. However `IPlugProcessor` already provides the overridable `OnReset` function so it could simply forward `clap::helpers::Plugin::reset` on to `IPlugProcessor::reset` if the semantics are correct.

`clap_plugin.reset` is called in the audio thread and is strictly for resetting the DSP back to its initial state (e.g. clearing reverb buffers back to zero). It's not for reallocating buffers in response to the sample rate changing. So I'm not sure if this maps directly to `IPlugProcessor::OnReset` or not.

**Comment by AlexHarker** [comment]
Yes this maps directly. I'm not sure if this existed when I completed IPlugCLAP or not, but for normal operation OnReset() is getting called in activate() so DSP resets should be correct.

However, if there are other cases where a DAW might want to directly call reset() then we can implement this with a direct mapping. Happy to look at a PR.

**Comment by AlexHarker** [comment]
@colugomusic  - please check to see if this branch addresses this for you:

https://github.com/iPlug2/iPlug2/tree/clap-fixes

**Comment by AlexHarker** [comment]
Now implemented by #1243
